sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/models/KeyVaultRoleDefinitionType.java

```diff
@@ -1,30 +1,42 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
 
 package com.azure.security.keyvault.administration.models;
 
 import com.azure.core.util.ExpandableStringEnum;
-import com.fasterxml.jackson.annotation.JsonCreator;
+import java.util.Collection;
 
-/**
- * Defines values for {@link KeyVaultRoleDefinitionType}.
- */
+/** The role definition type. */
 public final class KeyVaultRoleDefinitionType extends ExpandableStringEnum<KeyVaultRoleDefinitionType> {
-    /**
-     * Static value Microsoft.Authorization/roleDefinitions for {@link KeyVaultRoleDefinitionType}.
-     */
+    /** Static value Microsoft.Authorization/roleDefinitions for KeyVaultRoleDefinitionType. */
     public static final KeyVaultRoleDefinitionType MICROSOFT_AUTHORIZATION_ROLE_DEFINITIONS =
-        fromString("Microsoft.Authorization/roleDefinitions");
+            fromString("Microsoft.Authorization/roleDefinitions");
 
     /**
-     * Creates or finds a {@link KeyVaultRoleDefinitionType} from its string representation.
+     * Creates a new instance of KeyVaultRoleDefinitionType value.
      *
-     * @param name A name to look for.
+     * @deprecated Use the {@link #fromString(String)} factory method.
+     */
+    @Deprecated
+    public KeyVaultRoleDefinitionType() {}
+
+    /**
+     * Creates or finds a KeyVaultRoleDefinitionType from its string representation.
      *
-     * @return The corresponding {@link KeyVaultRoleDefinitionType}.
+     * @param name a name to look for.
+     * @return the corresponding KeyVaultRoleDefinitionType.
      */
-    @JsonCreator
     public static KeyVaultRoleDefinitionType fromString(String name) {
         return fromString(name, KeyVaultRoleDefinitionType.class);
     }
+
+    /**
+     * Gets known KeyVaultRoleDefinitionType values.
+     *
+     * @return known KeyVaultRoleDefinitionType values.
+     */
+    public static Collection<KeyVaultRoleDefinitionType> values() {
```

Comment:
This and other classes where `values()` was added are going to require a minor version bump, right?